play-client/src/main/scala/endpoints/EndpointPlayClient.scala

```diff
@@ -28,21 +28,20 @@ class EndpointPlayClient(wsClient: WSClient)(implicit ec: ExecutionContext) exte
   /**
     * A function that, given an `A` information and a `WSRequest`, eventually returns a `WSResponse`
     */
-  type RequestEntity[A] = (A, WSRequest) => Future[WSResponse]
+  type RequestEntity[A] = (A, WSRequest) => WSRequest
+
+  override val emptyRequestEntity: RequestEntity[Unit] = { case (_, req) => req }
 
-  def get[A, B](url: Url[A], headers: RequestHeaders[B])(implicit tupler: Tupler[A, B]): Request[tupler.Out] =
-    (ab: tupler.Out) => {
-      val (a, b) = tupler.unapply(ab)
-      val wsRequest = wsClient.url(url.encode(a))
-      headers(b, wsRequest).get()
-    }
 
-  def post[A, B, C, AB](url: Url[A], entity: RequestEntity[B], headers: RequestHeaders[C])(implicit tuplerAB: Tupler.Aux[A, B, AB], tuplerABC: Tupler[AB, C]): Request[tuplerABC.Out] =
+  override def request[A, B, C, AB](
```

Comment:
> So you are safe from cases when you want to override something but in the end you don't.

IMHO, this is not really a problem because in such a case at some point the compiler will complain that some method is not implemented.

On the other hand, if you always put the `override` keyword you can unintentionally override a method that was already implemented.

I wish I had a `@implement` annotation that would emit an error if you make a mistake in the type signature of the method you implement… That would be far better than waiting for the point where the compiler complains that there are unimplemented members.